dkg/nodesigs.go

```diff
@@ -49,33 +62,94 @@ func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
 	return ret
 }
 
+// lockHash returns the lock hash byte array from lockHashCh.
+// Once the data has been received, it stays cached in lockHashData.
+func (n *nodeSigBcast) lockHash(ctx context.Context) ([]byte, error) {
+	n.lhLock.Lock()
+	defer n.lhLock.Unlock()
+
+	if len(n.lockHashData) != 0 {
+		return n.lockHashData, nil
+	}
+
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case n.lockHashData = <-n.lockHashCh:
+		return n.lockHashData, nil
+	}
+}
+
+// allSigs returns true if all the node signatures have been received.
+func (n *nodeSigBcast) allSigs() bool {
+	n.osLock.Lock()
+	defer n.osLock.Unlock()
+
+	for _, sig := range n.otherSigs {
+		if sig == nil {
```

Comment:
always do `len(sig) == 0` comparison when check for empty slices, not `== nil`.

Reply:
yeah that's a leftover from a refactor, originally it was a slice of pointers then moved to array of bytes